### PR TITLE
[SCB-1497]Support for skywalking

### PR DESCRIPTION
**Pull request by surechen**
for jira issue SCB-1497

Signed-off-by: surechen <chenshuo17@huawei.com>

### CODE REVIEW

File: conf/mesher.yaml [hunk 1]
```diff
@@ -21,3 +21,7 @@ admin: #admin API
 #    match:
 #      status: 200
 #      body: ok
+#appPerfManage:
```
Comment (tianxiaoliang) [inline conf/mesher.yaml]: appPerfManage--->apm

File: conf/mesher.yaml [hunk 2]
```diff
@@ -21,3 +21,7 @@ admin: #admin API
...
+#  apmName: skywalking
+#  enable: true
+#  serverUri: 192.168.88.64:11800
```
Comment (tianxiaoliang) [inline conf/mesher.yaml]: apm:
  tracing:
    serverUri: 192.168.88.64:11800 
    enable: true

File: go.mod [hunk 3]
```diff
@@ -9,44 +9,43 @@ require (
...
-	github.com/google/gofuzz v0.0.0-20170612174753-24818f796faf // indirect
 	github.com/gopherjs/gopherjs v0.0.0-20181103185306-d547d1d9531e // indirect
-	github.com/imdario/mergo v0.3.6 // indirect
 	github.com/lyft/protoc-gen-validate v0.0.11 // indirect
 	github.com/patrickmn/go-cache v2.1.0+incompatible
+	github.com/pierrec/lz4 v2.3.0+incompatible // indirect
 	github.com/prometheus/client_golang v0.9.1
 	github.com/prometheus/client_model v0.0.0-20190115171406-56726106282f
 	github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1 // indirect
+	github.com/rcrowley/go-metrics v0.0.0-20190826022208-cac0b30c2563 // indirect
```
Comment (tianxiaoliang) [inline go.mod]: why introduce this lib

File: proxy/handler/skywalking_handler.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+func (sp *SkyWalkingProviderHandler) Handle(chain *handler.Chain, i *invocation.Invocation, cb invocation.ResponseCallBack) {
+	openlogging.GetLogger().Debugf("SkyWalkingProviderHandler begin. inv:%#v", *i)
+	span, _, err := skywalking.CreateEntrySpan(i)
+	if err != nil {
+		openlogging.GetLogger().Errorf("CreateEntrySpan error:%s", err.Error())
+	}
+	chain.Next(i, func(r *invocation.Response) (err error) {
+		err = cb(r)
+		switch i.Protocol {
+		case common.ProtocolRest:
```
Comment (tianxiaoliang) [inline proxy/handler/skywalking_handler.go]: why only support http protocol?

File: conf/mesher.yaml (hunk 1 above)
Comment (surechen) [reply conf/mesher.yaml]: Edit config struct apm.

File: conf/mesher.yaml (hunk 2 above)
Comment (surechen) [reply conf/mesher.yaml]: Edit config struct of apm.

File: go.mod (hunk 3 above)
Comment (surechen) [reply go.mod]: Misintroduction in Code Reading. Delete

File: proxy/handler/skywalking_handler.go (hunk 4 above)
Comment (surechen) [reply proxy/handler/skywalking_handler.go]: Edit

**Comment by surechen** [comment]
Reopen a PR for easy reading

**Comment by surechen** [comment]
Reopen a PR for easy reading

**Pull request closed without merging**